### ISSUE TITLE
[UT] FE UT timeout in GlobalTransactionMgrTest.testRetryCommitOnRateLimitExceededTimeout

### ISSUE THREAD

**Issue by kevincai** [StarRocks/starrocks#41882]
```
2024-02-28 10:35:40
Full thread dump OpenJDK 64-Bit Server VM (11.0.21+9-adhoc.root.jdk11u mixed mode):

Threads class SMR info:
_java_thread_list=0x00007f8748491f30, length=16, elements={
0x00007f87b8023800, 0x00007f87b8388000, 0x00007f87b838c000, 0x00007f87b83a5000,
0x00007f87b83ab000, 0x00007f87b83ad000, 0x00007f87b83af800, 0x00007f87b83b1800,
0x00007f87b83e2000, 0x00007f87b9a59800, 0x00007f87b9abb800, 0x00007f8708010800,
0x00007f87ba630800, 0x00007f8768001000, 0x00007f86f4001000, 0x00007f87ba121000
}

"main" #1 prio=5 os_prio=0 cpu=459202.17ms elapsed=2722.43s tid=0x00007f87b8023800 nid=0xebef runnable  [0x00007f87c0e91000]
   java.lang.Thread.State: RUNNABLE
	at java.util.HashMap.putVal(java.base@11.0.21/HashMap.java:631)
	at java.util.HashMap.put(java.base@11.0.21/HashMap.java:608)
	at java.util.HashSet.add(java.base@11.0.21/HashSet.java:220)
	at net.bytebuddy.dynamic.scaffold.MethodGraph$Compiler$Default$Key$Harmonized.detach(MethodGraph.java:1022)
	at net.bytebuddy.dynamic.scaffold.MethodGraph$Compiler$Default$Key$Store.asGraph(MethodGraph.java:1253)
	at net.bytebuddy.dynamic.scaffold.MethodGraph$Compiler$Default.compile(MethodGraph.java:660)
	at net.bytebuddy.dynamic.scaffold.MethodGraph$Compiler$AbstractBase.compile(MethodGraph.java:519)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.isOverridden(MockMethodAdvice.java:189)
	at com.starrocks.transaction.TransactionState.getTableIdList(TransactionState.java:667)
	at com.starrocks.transaction.GlobalTransactionMgr.commitTransactionUnderDatabaseWLock(GlobalTransactionMgr.java:402)
	at com.starrocks.transaction.GlobalTransactionMgr.retryCommitOnRateLimitExceeded(GlobalTransactionMgr.java:367)
	at java.lang.invoke.LambdaForm$DMH/0x000000080057c040.invokeVirtual(java.base@11.0.21/LambdaForm$DMH)
	at java.lang.invoke.LambdaForm$MH/0x0000000800589c40.invoke(java.base@11.0.21/LambdaForm$MH)
	at java.lang.invoke.LambdaForm$MH/0x000000080057e040.invoke(java.base@11.0.21/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.invokeExact_MT(java.base@11.0.21/Invokers$Holder)
	at java.lang.invoke.MethodHandle.invokeWithArguments(java.base@11.0.21/MethodHandle.java:710)
	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor$Dispatcher$ByteBuddy$eNiQd09p.invokeWithArguments(Unknown Source)
	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor.invoke(InstrumentationMemberAccessor.java:242)
	at org.mockito.internal.util.reflection.ModuleMemberAccessor.invoke(ModuleMemberAccessor.java:55)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.tryInvoke(MockMethodAdvice.java:314)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.access$500(MockMethodAdvice.java:64)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice$RealMethodCall.invoke(MockMethodAdvice.java:234)
	at org.mockito.internal.invocation.InterceptedInvocation.callRealMethod(InterceptedInvocation.java:142)
	at org.mockito.internal.stubbing.answers.CallsRealMethods.answer(CallsRealMethods.java:45)
	at org.mockito.Answers.answer(Answers.java:90)
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:111)
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:34)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:82)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.handle(MockMethodAdvice.java:134)
	at com.starrocks.transaction.GlobalTransactionMgr.retryCommitOnRateLimitExceeded(GlobalTransactionMgr.java:364)
	at com.starrocks.transaction.GlobalTransactionMgr.commitAndPublishTransaction(GlobalTransactionMgr.java:335)
	at java.lang.invoke.LambdaForm$DMH/0x000000080057fc40.invokeVirtual(java.base@11.0.21/LambdaForm$DMH)
	at java.lang.invoke.LambdaForm$MH/0x000000080057cc40.invoke(java.base@11.0.21/LambdaForm$MH)
	at java.lang.invoke.LambdaForm$MH/0x000000080057e040.invoke(java.base@11.0.21/LambdaForm$MH)
	at java.lang.invoke.Invokers$Holder.invokeExact_MT(java.base@11.0.21/Invokers$Holder)
	at java.lang.invoke.MethodHandle.invokeWithArguments(java.base@11.0.21/MethodHandle.java:710)
	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor$Dispatcher$ByteBuddy$eNiQd09p.invokeWithArguments(Unknown Source)
	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor.invoke(InstrumentationMemberAccessor.java:242)
	at org.mockito.internal.util.reflection.ModuleMemberAccessor.invoke(ModuleMemberAccessor.java:55)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.tryInvoke(MockMethodAdvice.java:314)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.access$500(MockMethodAdvice.java:64)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice$RealMethodCall.invoke(MockMethodAdvice.java:234)
	at org.mockito.internal.invocation.InterceptedInvocation.callRealMethod(InterceptedInvocation.java:142)
	at org.mockito.internal.stubbing.answers.CallsRealMethods.answer(CallsRealMethods.java:45)
	at org.mockito.Answers.answer(Answers.java:90)
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:111)
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:34)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:82)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.handle(MockMethodAdvice.java:134)
	at com.starrocks.transaction.GlobalTransactionMgr.commitAndPublishTransaction(GlobalTransactionMgr.java:334)
	at com.starrocks.transaction.GlobalTransactionMgrTest.lambda$testRetryCommitOnRateLimitExceededTimeout$0(GlobalTransactionMgrTest.java:873)
	at com.starrocks.transaction.GlobalTransactionMgrTest$$Lambda$500/0x00000008005c0040.run(Unknown Source)
	at org.junit.Assert.assertThrows(Assert.java:1001)
	at org.junit.Assert.assertThrows(Assert.java:981)
	at com.starrocks.transaction.GlobalTransactionMgrTest.testRetryCommitOnRateLimitExceededTimeout(GlobalTransactionMgrTest.java:873)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@11.0.21/Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@11.0.21/NativeMethodAccessorImpl.java:62)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.21/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.21/Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at mockit.integration.junit4.JUnit4TestRunnerDecorator.executeTestMethod(JUnit4TestRunnerDecorator.java:157)
	at mockit.integration.junit4.JUnit4TestRunnerDecorator.invokeExplosively(JUnit4TestRunnerDecorator.java:71)
	at mockit.integration.junit4.FakeFrameworkMethod.invokeExplosively(FakeFrameworkMethod.java:29)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$202/0x0000000800190840.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

"Reference Handler" #2 daemon prio=10 os_prio=0 cpu=19.97ms elapsed=2722.32s tid=0x00007f87b8388000 nid=0xec02 runnable  [0x00007f87853fa000]
   java.lang.Thread.State: RUNNABLE
	at java.lang.Object.notifyAll(java.base@11.0.21/Native Method)
	at java.lang.ref.ReferenceQueue.enqueue(java.base@11.0.21/ReferenceQueue.java:80)
	- locked <0x0000000702eda810> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.Reference.processPendingReferences(java.base@11.0.21/Reference.java:262)
	at java.lang.ref.Reference$ReferenceHandler.run(java.base@11.0.21/Reference.java:213)

"Finalizer" #3 daemon prio=8 os_prio=0 cpu=0.77ms elapsed=2722.32s tid=0x00007f87b838c000 nid=0xec03 in Object.wait()  [0x00007f87852f9000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(java.base@11.0.21/Native Method)
	- waiting on <0x000000070583fb98> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.21/ReferenceQueue.java:155)
	- waiting to re-lock in wait() <0x000000070583fb98> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.21/ReferenceQueue.java:176)
	at java.lang.ref.Finalizer$FinalizerThread.run(java.base@11.0.21/Finalizer.java:170)

"Signal Dispatcher" #4 daemon prio=9 os_prio=0 cpu=0.79ms elapsed=2722.29s tid=0x00007f87b83a5000 nid=0xec06 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #5 daemon prio=9 os_prio=0 cpu=0.11ms elapsed=2722.28s tid=0x00007f87b83ab000 nid=0xec07 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #6 daemon prio=9 os_prio=0 cpu=13780.87ms elapsed=2722.28s tid=0x00007f87b83ad000 nid=0xec08 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

"C1 CompilerThread0" #16 daemon prio=9 os_prio=0 cpu=1392.99ms elapsed=2722.28s tid=0x00007f87b83af800 nid=0xec09 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

"Sweeper thread" #21 daemon prio=9 os_prio=0 cpu=23.98ms elapsed=2722.27s tid=0x00007f87b83b1800 nid=0xec0a runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Common-Cleaner" #22 daemon prio=8 os_prio=0 cpu=3.36ms elapsed=2722.07s tid=0x00007f87b83e2000 nid=0xec12 in Object.wait()  [0x00007f87847ee000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(java.base@11.0.21/Native Method)
	- waiting on <no object reference available>
	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.21/ReferenceQueue.java:155)
	- waiting to re-lock in wait() <0x0000000705704d68> (a java.lang.ref.ReferenceQueue$Lock)
	at jdk.internal.ref.CleanerImpl.run(java.base@11.0.21/CleanerImpl.java:148)
	at java.lang.Thread.run(java.base@11.0.21/Thread.java:829)
	at jdk.internal.misc.InnocuousThread.run(java.base@11.0.21/InnocuousThread.java:161)

"surefire-forkedjvm-command-thread" #24 daemon prio=5 os_prio=0 cpu=59.40ms elapsed=2719.82s tid=0x00007f87b9a59800 nid=0xecb4 runnable  [0x00007f873f3f4000]
   java.lang.Thread.State: RUNNABLE
	at java.io.FileInputStream.readBytes(java.base@11.0.21/Native Method)
	at java.io.FileInputStream.read(java.base@11.0.21/FileInputStream.java:279)
	at java.io.BufferedInputStream.fill(java.base@11.0.21/BufferedInputStream.java:252)
	at java.io.BufferedInputStream.read(java.base@11.0.21/BufferedInputStream.java:271)
	- locked <0x00000007060ab368> (a java.io.BufferedInputStream)
	at java.io.DataInputStream.readInt(java.base@11.0.21/DataInputStream.java:392)
	at org.apache.maven.surefire.booter.MasterProcessCommand.decode(MasterProcessCommand.java:115)
	at org.apache.maven.surefire.booter.CommandReader$CommandRunnable.run(CommandReader.java:391)
	at java.lang.Thread.run(java.base@11.0.21/Thread.java:829)

"surefire-forkedjvm-ping-30s" #25 daemon prio=5 os_prio=0 cpu=768.86ms elapsed=2719.59s tid=0x00007f87b9abb800 nid=0xecc8 runnable  [0x00007f873ebe4000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.21/Native Method)
	- parking to wait for  <0x0000000706661800> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.21/LockSupport.java:234)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.21/AbstractQueuedSynchronizer.java:2123)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.21/ScheduledThreadPoolExecutor.java:1182)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.21/ScheduledThreadPoolExecutor.java:899)
	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.21/ThreadPoolExecutor.java:1054)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.21/ThreadPoolExecutor.java:1114)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.21/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.21/Thread.java:829)

"process reaper" #26 daemon prio=10 os_prio=0 cpu=641.24ms elapsed=2719.48s tid=0x00007f8708010800 nid=0xecd5 waiting on condition  [0x00007f87c0d21000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.21/Native Method)
	- parking to wait for  <0x0000000706666b00> (a java.util.concurrent.SynchronousQueue$TransferStack)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.21/LockSupport.java:234)
	at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(java.base@11.0.21/SynchronousQueue.java:462)
	at java.util.concurrent.SynchronousQueue$TransferStack.transfer(java.base@11.0.21/SynchronousQueue.java:361)
	at java.util.concurrent.SynchronousQueue.poll(java.base@11.0.21/SynchronousQueue.java:937)
	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.21/ThreadPoolExecutor.java:1053)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.21/ThreadPoolExecutor.java:1114)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.21/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.21/Thread.java:829)

"Metric-Timer-Pool-0" #108 daemon prio=5 os_prio=0 cpu=34.26ms elapsed=2711.36s tid=0x00007f87ba630800 nid=0xf00a waiting on condition  [0x00007f873e7e2000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.21/Native Method)
	- parking to wait for  <0x000000070647f778> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.21/LockSupport.java:234)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.21/AbstractQueuedSynchronizer.java:2123)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.21/ScheduledThreadPoolExecutor.java:1182)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.21/ScheduledThreadPoolExecutor.java:899)
	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.21/ThreadPoolExecutor.java:1054)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.21/ThreadPoolExecutor.java:1114)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.21/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.21/Thread.java:829)

"Attach Listener" #515 daemon prio=9 os_prio=0 cpu=1.62ms elapsed=2704.86s tid=0x00007f8768001000 nid=0xf2e3 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"pool-324-thread-1" #678 prio=5 os_prio=0 cpu=41.62ms elapsed=2696.36s tid=0x00007f86f4001000 nid=0xf67b runnable  [0x00007f86bb1fc000]
   java.lang.Thread.State: RUNNABLE
	at jdk.internal.misc.Unsafe.park(java.base@11.0.21/Native Method)
	- parking to wait for  <0x000000070647f078> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.21/LockSupport.java:234)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.21/AbstractQueuedSynchronizer.java:2123)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.21/ScheduledThreadPoolExecutor.java:1182)
	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.21/ScheduledThreadPoolExecutor.java:899)
	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.21/ThreadPoolExecutor.java:1054)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.21/ThreadPoolExecutor.java:1114)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.21/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.21/Thread.java:829)

"Thread-948" #1002 prio=5 os_prio=0 cpu=0.23ms elapsed=2691.16s tid=0x00007f87ba121000 nid=0xf8a2 waiting on condition  [0x00007f8784cf3000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.21/Native Method)
	- parking to wait for  <0x00000007063d1b18> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.21/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.21/AbstractQueuedSynchronizer.java:2081)
	at java.util.concurrent.ArrayBlockingQueue.take(java.base@11.0.21/ArrayBlockingQueue.java:417)
	at com.starrocks.utframe.UtFrameUtils$PseudoJournalReplayer$1.run(UtFrameUtils.java:1019)
	at java.lang.Thread.run(java.base@11.0.21/Thread.java:829)

"VM Thread" os_prio=0 cpu=21499.43ms elapsed=2722.34s tid=0x00007f87b8380000 nid=0xec00 runnable  

"GC Thread#0" os_prio=0 cpu=1993703.70ms elapsed=2722.43s tid=0x00007f87b803d000 nid=0xebf0 runnable  

"GC Thread#1" os_prio=0 cpu=1994675.31ms elapsed=2721.04s tid=0x00007f877c002800 nid=0xec66 runnable  

"GC Thread#2" os_prio=0 cpu=1993305.94ms elapsed=2721.03s tid=0x00007f877c004800 nid=0xec67 runnable  

"GC Thread#3" os_prio=0 cpu=1996896.98ms elapsed=2721.01s tid=0x00007f877c006000 nid=0xec68 runnable  

"GC Thread#4" os_prio=0 cpu=1993061.17ms elapsed=2721.01s tid=0x00007f877c007800 nid=0xec69 runnable  

"GC Thread#5" os_prio=0 cpu=1993828.45ms elapsed=2721.00s tid=0x00007f877c009000 nid=0xec6a runnable  

"GC Thread#6" os_prio=0 cpu=1995829.98ms elapsed=2720.97s tid=0x00007f877c00a800 nid=0xec6c runnable  

"GC Thread#7" os_prio=0 cpu=1995985.15ms elapsed=2720.97s tid=0x00007f877c00c800 nid=0xec6d runnable  

"GC Thread#8" os_prio=0 cpu=1994394.72ms elapsed=2720.96s tid=0x00007f877c00e000 nid=0xec6e runnable  

"GC Thread#9" os_prio=0 cpu=1994470.35ms elapsed=2720.95s tid=0x00007f877c00f800 nid=0xec6f runnable  

"GC Thread#10" os_prio=0 cpu=1994631.05ms elapsed=2720.94s tid=0x00007f877c011000 nid=0xec70 runnable  

"GC Thread#11" os_prio=0 cpu=1994767.00ms elapsed=2712.70s tid=0x00007f877c015000 nid=0xefa2 runnable  

"GC Thread#12" os_prio=0 cpu=1994996.54ms elapsed=2712.69s tid=0x00007f877c016800 nid=0xefa3 runnable  

"GC Thread#13" os_prio=0 cpu=1993286.34ms elapsed=2712.68s tid=0x00007f877c018000 nid=0xefa4 runnable  

"GC Thread#14" os_prio=0 cpu=1994676.03ms elapsed=2712.67s tid=0x00007f877c019800 nid=0xefa7 runnable  

"GC Thread#15" os_prio=0 cpu=1995245.02ms elapsed=2712.67s tid=0x00007f877c01b000 nid=0xefa8 runnable  

"GC Thread#16" os_prio=0 cpu=1997134.61ms elapsed=2712.65s tid=0x00007f877c01c800 nid=0xefab runnable  

"GC Thread#17" os_prio=0 cpu=1994699.67ms elapsed=2712.63s tid=0x00007f877c01e800 nid=0xefae runnable  

"GC Thread#18" os_prio=0 cpu=1994107.27ms elapsed=2682.41s tid=0x00007f877c02a800 nid=0xfc15 runnable  

"GC Thread#19" os_prio=0 cpu=1994079.57ms elapsed=2682.41s tid=0x00007f877c1bb800 nid=0xfc17 runnable  

"GC Thread#20" os_prio=0 cpu=1991496.57ms elapsed=2682.41s tid=0x00007f877c1bc800 nid=0xfc18 runnable  

"GC Thread#21" os_prio=0 cpu=1993242.45ms elapsed=2682.41s tid=0x00007f877c1bd800 nid=0xfc19 runnable  

"GC Thread#22" os_prio=0 cpu=1997639.74ms elapsed=2682.41s tid=0x00007f877c1be800 nid=0xfc1a runnable  

"G1 Main Marker" os_prio=0 cpu=96.57ms elapsed=2722.41s tid=0x00007f87b809e800 nid=0xebf7 runnable  

"G1 Conc#0" os_prio=0 cpu=190075.74ms elapsed=2722.40s tid=0x00007f87b80a0800 nid=0xebf9 runnable  

"G1 Conc#1" os_prio=0 cpu=192668.62ms elapsed=2716.59s tid=0x00007f8788001000 nid=0xee06 runnable  

"G1 Conc#2" os_prio=0 cpu=191045.20ms elapsed=2716.58s tid=0x00007f8788002000 nid=0xee07 runnable  

"G1 Conc#3" os_prio=0 cpu=193157.72ms elapsed=2716.58s tid=0x00007f8788003000 nid=0xee0a runnable  

"G1 Conc#4" os_prio=0 cpu=187981.52ms elapsed=2716.57s tid=0x00007f8788004000 nid=0xee0b runnable  

"G1 Conc#5" os_prio=0 cpu=196690.86ms elapsed=2716.57s tid=0x00007f8788005000 nid=0xee0c runnable  

"G1 Refine#0" os_prio=0 cpu=188.08ms elapsed=2722.37s tid=0x00007f87b82b1800 nid=0xebfc runnable  

"G1 Refine#1" os_prio=0 cpu=102.74ms elapsed=2509.16s tid=0x00007f8780001000 nid=0x13e09 runnable  

"G1 Refine#2" os_prio=0 cpu=44.72ms elapsed=2509.15s tid=0x00007f870033d000 nid=0x13e0e runnable  

"G1 Refine#3" os_prio=0 cpu=19.33ms elapsed=2509.13s tid=0x00007f86d400f800 nid=0x13e15 runnable  

"G1 Refine#4" os_prio=0 cpu=3.54ms elapsed=2503.87s tid=0x00007f86d8029000 nid=0x13fc0 runnable  

"G1 Refine#5" os_prio=0 cpu=0.17ms elapsed=2498.79s tid=0x00007f86a8013000 nid=0x14197 runnable  

"G1 Refine#6" os_prio=0 cpu=0.05ms elapsed=2491.83s tid=0x00007f86b026f000 nid=0x1440f runnable  

"G1 Young RemSet Sampling" os_prio=0 cpu=273.60ms elapsed=2722.37s tid=0x00007f87b82b3000 nid=0xebfd runnable  
"VM Periodic Task Thread" os_prio=0 cpu=828.50ms elapsed=2720.11s tid=0x00007f87b9a22000 nid=0xeca6 waiting on condition  

JNI global refs: 32, weak refs: 0

```

**Comment by kevincai** [comment]
GC threads use a lot of cpu times, sound to me GC issue? possibly jvm memory is not enough?

**Comment by kevincai** [comment]
some more information about this issue.

```
 S0C    S1C    S0U    S1U      EC       EU        OC         OU       MC     MU    CCSC   CCSU   YGC     YGCT    FGC    FGCT    CGC    CGCT     GCT   
 0.0    0.0    0.0    0.0     0.0      0.0    4194304.0  4194205.4  59136.0 57362.5 5888.0 5306.1    813   17.760  230   853.833  169     0.642  872.236
 0.0    0.0    0.0    0.0     0.0      0.0    4194304.0  4194240.5  59136.0 57362.5 5888.0 5306.1    814   17.763  231   857.663  169     0.642  876.069
 0.0    0.0    0.0    0.0     0.0      0.0    4194304.0  4194240.5  59136.0 57362.5 5888.0 5306.1    814   17.763  231   857.663  169     0.642  876.069
 0.0    0.0    0.0    0.0     0.0      0.0    4194304.0  4194240.5  59136.0 57362.5 5888.0 5306.1    814   17.763  231   857.663  169     0.642  876.069
 0.0    0.0    0.0    0.0     0.0      0.0    4194304.0  4194240.5  59136.0 57362.5 5888.0 5306.1    814   17.763  231   857.663  169     0.642  876.069
 0.0    0.0    0.0    0.0     0.0      0.0    4194304.0  4194277.9  59136.0 57362.5 5888.0 5306.1    815   17.766  232   861.555  169     0.642  879.963
 0.0    0.0    0.0    0.0     0.0      0.0    4194304.0  4194277.9  59136.0 57362.5 5888.0 5306.1    815   17.766  232   861.555  169     0.642  879.963
```

**Comment by kevincai** [comment]
```
 num     #instances         #bytes  class name (module)
-------------------------------------------------------
   1:       8186440      720406720  java.lang.reflect.Method (java.base@11.0.21)
   2:      16369168      523813376  java.lang.ref.WeakReference (java.base@11.0.21)
   3:       8186465      392950320  java.lang.invoke.MemberName (java.base@11.0.21)
   4:       8184371      392849808  org.mockito.internal.invocation.InterceptedInvocation
   5:       8184371      261899872  java.lang.StackFrameInfo (java.base@11.0.21)
   6:       8184371      261899872  org.mockito.internal.creation.bytebuddy.MockMethodAdvice$RealMethodCall
   7:      16368742      261899872  org.mockito.internal.invocation.mockref.MockWeakReference
   8:       8194211      223427968  [Ljava.lang.Object; (java.base@11.0.21)
   9:       8186912      222681696  [Ljava.lang.Class; (java.base@11.0.21)
  10:       8185717      196457208  java.util.LinkedList$Node (java.base@11.0.21)
  11:       8184371      196424904  org.mockito.internal.creation.DelegatingMethod
  12:       8184371      196424904  org.mockito.internal.debugging.Java9PlusLocationImpl
  13:       8184371      130949936  org.mockito.internal.debugging.Java9PlusLocationImpl$MetadataShim
  14:       4913503      117924072  java.lang.Long (java.base@11.0.21)
  15:         47819       86159480  [B (java.base@11.0.21)
  16:       4910614       78569824  org.mockito.internal.invocation.StubInfoImpl
  17:          4439       11938448  [I (java.base@11.0.21)
  18:         45030        1080720  java.lang.String (java.base@11.0.21)
  19:          8183         995776  java.lang.Class (java.base@11.0.21)
  20:         21891         700512  java.util.concurrent.ConcurrentHashMap$Node (java.base@11.0.21)
  21:          6568         630528  com.starrocks.catalog.AggregateFunction
  22:         15859         507488  java.util.HashMap$Node (java.base@11.0.21)
  23:          3448         303424  com.starrocks.catalog.ScalarFunction
  24:          2674         260960  [Z (java.base@11.0.21)
  25:         10073         246296  [Lcom.starrocks.catalog.Type;
  26:         10074         241776  com.starrocks.analysis.FunctionName
  27:          1841         224648  [Ljava.util.HashMap$Node; (java.base@11.0.21)
  28:           291         181952  [Ljava.util.concurrent.ConcurrentHashMap$Node; (java.base@11.0.21)
  29:         10000         160000  java.lang.Object (java.base@11.0.21)
```

**Comment by kevincai** [comment]
#42970

**Comment by kevincai** [comment]
should be fixed in #42970
